DataProducts/inc/STMChannel.hh

```diff
@@ -0,0 +1,129 @@
+#ifndef DataProducts_STMChannel_hh
+#define DataProducts_STMChannel_hh
+//
+// An enum-matched-to-names class for generator Id's.
+// based on GenId.hh
+//
+
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace mu2e {
+
+  class STMChannel {
+
+  public:
+
+    // Need to keep the enum and the _name member in sync.
+    // Also, restrict to 8-bits because we will never need that many
+    // and this may be in every STMDigi
+    enum enum_type : int8_t { HPGe, LaBr, unknown, lastEnum };
+
+#ifndef SWIG
+    // Keep this in sync with the enum. Used in STMChannel.cc
+#define STMCHANNEL_NAMES \
+    "HPGe", "LaBr", "unknown"
+#endif
+
+  public:
+
+    // The most important c'tor and accessor methods are first.
+    STMChannel( enum_type id):
+      _id(id)
+    {}
+
+    enum_type id() const { return _id;}
+
+    const std::string name() const {
+      return std::string( _name[_id] );
+    }
+
+    // ROOT requires a default c'tor.
+    STMChannel():
+      _id(unknown){
+    }
+
+    virtual ~STMChannel(){}
+
+    bool operator==(const STMChannel g) const{
+      return ( _id == g._id );
+    }
+
+    bool operator==(const STMChannel::enum_type g) const{
+      return ( _id == g );
+    }
+
+    bool operator!=(const STMChannel::enum_type g) const{
+      return ( _id != g );
+    }
+
+    bool operator!=(const STMChannel g) const{
+      return ( _id != g._id );
+    }
+
+    // Accessor for the version.
+    static int version() { return _version; }
+
+    // Static version of the name method.
+    static const std::string name( enum_type id ){
+      return std::string( _name[id] );
+    }
+
+    // Check validity of an Id. Unknown is defined to be valid.
+    static bool isValid( enum_type id){
+      if ( id <  unknown  ) return false;
```

Comment:
This needs maintenance because of the change in the numeric value of unknown.  The right test is <0 or >=unknown returns false.

Reply:
Thanks, Rob. Done :)

Reply:
Thanks.